src/web/app/pages-instructor/instructor-home-page/instructor-home-page.component.ts

```diff
@@ -207,6 +319,24 @@ export class InstructorHomePageComponent extends InstructorSessionModalPageCompo
             this.statusMessageService.showErrorToast(resp.error.message);
           },
         });
+    this.courseService.getAllCoursesAsInstructor('archived').subscribe({
+      next: (resp: Courses) => {
+        this.allCoursesList.push(...resp.courses);
+      },
+      error: (resp: ErrorMessageOutput) => {
+        this.hasCoursesLoadingFailed = true;
+        this.statusMessageService.showErrorToast(resp.error.message);
+      },
+    });
+    this.courseService.getAllCoursesAsInstructor('softDeleted').subscribe({
+      next: (resp: Courses) => {
+        this.allCoursesList.push(...resp.courses);
+      },
+      error: (resp: ErrorMessageOutput) => {
+        this.hasCoursesLoadingFailed = true;
+        this.statusMessageService.showErrorToast(resp.error.message);
+      },
+    });
```

Comment:
Hmm can you help me understand what role this plays? Sorry if I missed something - to my knowledge, the course copy option for the home page works even without it

Reply:
So I based these lines off of [these ones](https://github.com/TEAMMATES/teammates/blob/396dff66914dc50ec7bcb1f11f4610175b4e5f4c/src/web/app/pages-instructor/instructor-courses-page/instructor-courses-page.component.ts#L162-L215), but just the part where they are adding the archived and softDeleted courses to the allCoursesList.

It is my understanding that after the list is sent to the copyCourseModalComponent here:
https://github.com/TEAMMATES/teammates/blob/e25399a2b20cb9cd3df75136b0c81a608dc59cd4/src/web/app/pages-instructor/instructor-home-page/instructor-home-page.component.ts#L163
 
the list is then referred to as this.allCourses, which is then used to check whether the new course Id is conflicting here: 
https://github.com/TEAMMATES/teammates/blob/05160f36c266cd2a172b57aeaed8d01d475da809/src/web/app/components/copy-course-modal/copy-course-modal.component.ts#L83-L84

So I thought it would be best that the new course's Id shouldn't conflict even with archived or softDeleted courses. But please let me know if I shouldn't worry about that check and I'll remove that.

Reply:
Ah yup that makes sense! Thanks for explaining (: